ipv8/configuration_pydantic.py

```diff
@@ -0,0 +1,402 @@
+    and this can be reflected when making birds (``make_bird``) by configuration:
+
+    |
+
+    ::
+
+        @dataclass
+        class DuckType(ABC):  # Abstract: a DuckType is not a bird, it just quacks
+            quacks_per_second: int
+
+        @dataclass
+        class Pigeon(DuckType):
+            in_flock: bool  # A group of pigeons is called a flock, not a murder
+
+        @dataclass
+        class Crow(DuckType):
+            in_murder: bool  # A group of crows is called a murder, not a flock
+
+        class DuckTypeConfig(BaseModel, AbstractModel):
+            quacks_per_second: int
+
+        class PigeonConfig(DuckTypeConfig):
+            in_flock: bool
+
+        class CrowConfig(DuckTypeConfig):
+            in_murder: bool
+
+        def make_bird(bird_config: DuckTypeConfig) -> Optional[DuckType]:
+            '''
+            Our composable architecture needs something that quacks per second.
+            The user registers implementations:
+
+            composable_architecture.register(Crow)
+            composable_architecture.register(Pigeon)
+            etc.
+
+            We don't need defensive programming in the composable_architecture
+            if the user can't make a DuckTypeConfig in the first place!
+            '''
+            return composable_architecture.get_implementer(bird_config)
+
+    :returns: The mixin for pydantic BaseModel subclasses.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        """
+        If mixed-in to ``check`` the MRO is [child 1 .. n, check, .. non-AbstractModel .., AbstractModel, object]
+        If there are no children this becomes [check, .. non-AbstractModel .., AbstractModel, object]
+        """
+        check = cls.mro()[1:2] or [object]  # Still works if the mro is empty
+        if not issubclass(check[0], AbstractModel) and check is not AbstractModel:
+            # Try to give the caller some hints.
+            parent_module = sys.modules[cls.__module__]
+            possibilities = [cname for cname, cinstance in inspect.getmembers(parent_module)
+                             if inspect.isclass(cinstance) and cname != "AbstractModel"
+                             and issubclass(cinstance, cls) and cinstance is not cls]
+            raise NotImplementedError(f"{cls} is an abstract model and must be implemented by a subclass! "
+                                      + (f"Perhaps you want one of these: {possibilities}?" if possibilities else ""))
+        return super().__new__(cls)
+
+
+class BaseConfigurationClass(BaseModel):
+    """
+    Augment the ``BaseModel`` class with 1) dict cast unwrapping and 2) inherited validators.
+    """
+
+    def __iter__(self, *args, **kwargs):
+        """
+        Make ``dict(model)`` equal to ``model.dict()``.
+
+        Normally this function only only gives the child model references instead of unwrapping them.
```

Comment:
Typo: duplicated word
```suggestion
        Normally this function only gives the child model references instead of unwrapping them.
```